tests/testthat/test-create_query_data.R

```diff
@@ -528,19 +516,123 @@ test_that("basket_select Test 21: basket_select: error: neither name nor id spec
 })
 
 # basket_select: error: type is not specified ----
-## Test 22: basket_select: error: type is not specified ----
-test_that("basket_select Test 22: basket_select: error: type is not specified", {
+## Test 21: basket_select: error: type is not specified ----
+test_that("basket_select Test 21: basket_select: error: type is not specified", {
   expect_error(
     basket_select(id = 42, scope = "NARROW"),
     regexp = "argument \"type\" is missing, with no default",
     fixed = TRUE
   )
 })
 
-# format.basket_select ----
+# basket_select customized query defined by SMQs extra arguments ----
+get_smq_oth <- function(basket_select,
```

Comment:
The function doesn't use the extra arguments (unless I missed it). Thus the test can't check if the new functionality works.

Reply:
See lines 520-521 and 527-528.

Reply:
The extra arguments are set in the `basket_select()` calls but `get_smq_oth()` does not use them. For example if I use
```
  cq <- query(
    prefix = "CQ02",
    name = "Immune-Mediated Meningoencephalitis",
    definition = list(
      basket_select(
        name = "Noninfectious meningitis",
        scope = "NARROW",
        type = "smq"
      ),
      basket_select(
        name = "Noninfectious encephalitis",
        scope = "BROAD",
        type = "smq"
      )
    )
  )
```
I still get the same results from the `create_query_data()` call. I.e., the test would still pass even if `basket_select()` ignores the extra arguments and doesn't stores its values in the object.

Reply:
thanks @bundfuss, I have updated test.